functions/metadata/metadata_test.go

```diff
@@ -120,3 +120,52 @@ func TestUnmarshalJSON(t *testing.T) {
 		}
 	}
 }
+
+func TestMarshalJSON(t *testing.T) {
+	ts, err := time.Parse("2006-01-02T15:04:05Z07:00", "2019-11-04T23:01:10.112Z")
+	if err != nil {
+		t.Fatalf("Error parsing time: %v.", err)
+	}
+	var tests = []struct {
+		name     string
+		metadata Metadata
+		want     []byte
+	}{
+		{
+			name: "MetadataWithResource",
+			metadata: Metadata{
+				EventID:   "1234567",
+				Timestamp: ts,
+				EventType: "google.pubsub.topic.publish",
+				Resource: &Resource{
+					Service: "pubsub.googleapis.com",
+					Name:    "mytopic",
+					Type:    "type.googleapis.com/google.pubsub.v1.PubsubMessage",
+				},
+			},
+			want: []byte(`{"eventId":"1234567","timestamp":"2019-11-04T23:01:10.112Z","eventType":"google.pubsub.topic.publish","resource":{"service":"pubsub.googleapis.com","name":"mytopic","type":"type.googleapis.com/google.pubsub.v1.PubsubMessage"}}`),
+		},
+		{
+			name: "MetadataWithString",
+			metadata: Metadata{
+				EventID:   "1234567",
+				Timestamp: ts,
+				EventType: "google.pubsub.topic.publish",
+				Resource: &Resource{
+					RawPath: "projects/myproject/mytopic",
+				},
+			},
+			want: []byte(`{"eventId":"1234567","timestamp":"2019-11-04T23:01:10.112Z","eventType":"google.pubsub.topic.publish","resource":"projects/myproject/mytopic"}`),
+		},
+	}
+
+	for _, tc := range tests {
+		b, err := json.Marshal(&tc.metadata)
```

Comment:
I wonder if this test could marshal then unmarshal and check the input and output are `cmp.Equal`, rather than checking against a known string?